src/persistency/PersistentStore.ts

```diff
@@ -0,0 +1,66 @@
+import objhash from 'object-hash'
+import { PersistentStoreStrategy, Updater } from './types'
+
+export default class PersistentStore {
+  constructor(private strategy: PersistentStoreStrategy, private prefix: string = '') {
+    if (typeof prefix !== 'string') {
+      throw new Error(`Datastore: Invalid prefix: ${prefix}`)
+    }
+  }
+
+  public createNamespace(prefix: string): PersistentStore {
```

Comment:
a method in a class that returns an instance of itself? I don't get it

Reply:
With a different prefix